File: java-api/src/main/java/com/webmvc/javaapi/controller/MemberController.java
package com.webmvc.javaapi.controller;

import com.webmvc.javaapi.dto.request.MemberSaveRequest;
import com.webmvc.javaapi.dto.request.MemberUpdateRequest;
import com.webmvc.javaapi.dto.response.ApiResponse;
import com.webmvc.javaapi.dto.response.MemberResponse;
import com.webmvc.javaapi.service.MemberService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("member")
@RequiredArgsConstructor
public class MemberController {

    private final MemberService memberService;

    @PostMapping
    public ApiResponse<MemberResponse> saveMember(@RequestBody @Valid MemberSaveRequest memberSaveRequest) {
        MemberResponse responseDto = memberService.saveMember(memberSaveRequest);

        return new ApiResponse<>(responseDto);
    }

    @GetMapping
    public ApiResponse<Page<MemberResponse>> findMembers(
            Pageable pageable,
            @RequestParam(required = false) String account,
            @RequestParam(required = false) String name
    ) {
        Page<MemberResponse> members = memberService.findMembers(pageable, account, name);

        return new ApiResponse<>(members);
    }

    @GetMapping("{id}")
    public ApiResponse<MemberResponse> findMember(@PathVariable Long id) {
        MemberResponse responseDto = memberService.findMember(id);

        return new ApiResponse<>(responseDto);
    }

    @PatchMapping("{id}")
    public ApiResponse<MemberResponse> updateMember(
            @PathVariable Long id,
            @RequestBody @Valid MemberUpdateRequest memberUpdateRequest
    ) {
        MemberResponse responseDto = memberService.updateMember(id, memberUpdateRequest);

        return new ApiResponse<>(responseDto);
    }

    @DeleteMapping("{id}")
    public ApiResponse<Void> deleteMember(@PathVariable Long id) {
        memberService.deleteMember(id);

        return new ApiResponse<>();
    }
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/domain/member/controller/MemberController.kt
package spring.web.kotlin.domain.member.controller

import org.springframework.data.domain.Pageable
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*
import spring.web.kotlin.domain.member.dto.MemberSaveRequest
import spring.web.kotlin.domain.member.dto.MemberUpdateRequest
import spring.web.kotlin.domain.member.service.MemberService

@RestController
@RequestMapping("members")
class MemberController(
    private val memberService: MemberService,
) {
    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    fun saveMember(@RequestBody memberSaveRequest: MemberSaveRequest) =memberService.saveMember(memberSaveRequest)

    @GetMapping
    fun findMembers(
        pageable: Pageable,
        @RequestParam(required = false) account: String?,
        @RequestParam(required = false) name: String?,
    ) = memberService.findMembers(pageable, account, name)

    @GetMapping("{id}")
    fun findMember(@PathVariable id: Long) = memberService.findMember(id)

    @PatchMapping("{id}")
    fun updateMember(@PathVariable id: Long, @RequestBody memberUpdateRequest: MemberUpdateRequest) =
        memberService.patchMember(id, memberUpdateRequest)

    @DeleteMapping("{id}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    fun deleteMember(@PathVariable id: Long) {
        memberService.deleteMember(id)
    }
}
